### PR TITLE
Add support for ABAP CDS

### PR DESCRIPTION
**Pull request by FreHu**
Already tried in #4592, but the repo had issues with the license and build. Submitting again with a new repo (and color). The grammar is exactly the same, but the repo does not contain the incompatibly licensed snippets.

As previously discussed, I'm only including one of the extensions for popularity reasons. Hopefully this will improve in the future.

## Description
Hi! This PR adds support for the ABAP CDS language, which is used for defining what is basically database views on steroids. It goes hand in hand with the ABAP language, which is already recognized.

The language is not widely used outside of the SAP ecosystem, but even the public github contains a number of repositories now. Many more can be found on github enterprise servers.

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Aasddls+define+view
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
  https://github.com/hardyp/AbapToTheFuture03/tree/master/src/zmonsters_p02_transactional/zmonsters_c07_database
    - Sample license(s): MIT
  
  - [x] I have included a syntax highlighting grammar: [here](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Ffrehu%2Fabap-cds-grammar%2Fblob%2Fmaster%2Fsyntaxes%2Fcds.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fhardyp%2FAbapToTheFuture03%2Fblob%2Fmaster%2Fsrc%2Fzmonsters_p02_transactional%2Fzmonsters_c07_database%2Fzcds_monsters_parameters.ddls.asddls&code=%40AbapCatalog.sqlViewName%3A+%27ZV_MONSTERS_ASS%27%0D%0A%40AbapCatalog.compiler.compareFilter%3A+true%0D%0A%40AbapCatalog.preserveKey%3A+true%0D%0A%40AccessControl.authorizationCheck%3A+%23CHECK%0D%0A%40EndUserText.label%3A+%27Monster+CDS+View+with+Association%27%0D%0A%2F%2F+Listing+07.30+%3A+Coding+CDS+Views+using+Association%0D%0Adefine+view+Zcds_Monsters_Association%0D%0A++as+select+from+ztmonster_header+as+monster%0D%0A++association+%5B0..*%5D+to+ztmonster_pets+as+_Pet+on+monster.monster_number+%3D+_Pet.owner%0D%0A%7B%0D%0A++key+monster.monster_number+as+MonsterNumber%2C%0D%0A++key+_Pet.pet_number++++++++as+PetNumber%2C%0D%0A++++++_Pet.owner+++++++++++++as+Owner%2C%0D%0A++++++monster.name+++++++++++as+OwnerName%2C%0D%0A++++++_Pet.pet_name++++++++++as+Name%2C%0D%0A++++++_Pet.pet_type++++++++++as+Type%2C%0D%0A++++++_Pet.pet_species+++++++as+Species%2C%0D%0A++++++_Pet+%2F%2F+Make+association+public%0D%0A%7D%0D%0A)
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.
     - I believe asddls is a rare enough extension to prevent any conflicts.

## Notes

- To simplify legal matters, I would like to note that though I am a SAP employee, I am making this request as my own person in my spare time and not on behalf of SAP.

### CODE REVIEW

File: samples/ABAP CDS/zcds_monsters_association.ddls.asddls.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@AbapCatalog.sqlViewName: 'ZV_MONSTERS_ASS'
```
Comment (pchaigno) [inline samples/ABAP CDS/zcds_monsters_association.ddls.asddls.txt]: It looks like a `.txt` extension was added to this file by mistake.

**Comment by lildude** [comment]
> I've downloaded the 804 files associated to this extension using Harvester and found 104 repositories by 51 users. This is clearly low given our requirements.

Yeah, unfortunately still too low. Popularity for inclusion is based on GitHub.com stats as that's the only thing we can measure. It's probably best to add this to https://github.com/github/linguist/issues/4219 and monitor the growth.

**Comment by FreHu** [comment]
Alright I guess. I'll try to motivate people to host more of their code on github, but unfortunately, most devs within my reach simply can't make their code public.

This is purely philosophical rambling from my side as I'm aware you currently don't have a better metric to judge popularity by, but: 

I'm taking the "build it and they'll come" approach, but in this case, they need to come _before_ I can build it, which is a bit of a conundrum. SAP is used by major companies which can be very resistant to change and have yet to adopt proper versioning and open source practices, but somehow expect perfect tooling to be available before they move a finger. As github is almost the de facto standard place for sharing code, lack of support there can thwart adoption of the language.

Anyway, I'm sure this will resolve itself over time, I'd just like to make that time shorter.

**Comment by dellagustin** [comment]
Please keep in mind the github enterprise use case.
ABAP code is not often made open source, but I bet there are a number of companies already hosting their ABAP code on their enterprise instances.

**Comment by FreHu** [comment]
No, because `.cds` and `.hdbcds` are a different language - CAP CDS. This request adds ABAP CDS. 

https://github.wdf.sap.corp/pages/cap/cds/
Why we name languages so similarly we can't tell them apart is beyond me.

Also the usage has to be per extension, not in total.

**Comment by pchaigno** [comment]
Closing in favor of #4219. We'll monitor the popularity of the extensions there.

**Comment by lildude** [comment]
Good news, popularity of the `.asddls` extension is now sufficient for inclusion so I'm re-opening this PR.

**Review by lildude — APPROVED** [review]

**Pull request merged**